mdbenchmark/plot.py

```diff
@@ -0,0 +1,246 @@
+        elif module not in df_module_list:
+            console.error("The module {} does not exist in your data. Exiting.", module)
+        if len(temp_module_names) is not 0:
+            processed_module_names = processed_module_names + temp_module_names
+
+    if len(processed_module_names) is 0 and len(module_name) is not 0:
+        console.error("No modules or engines were detected in your data.")
+
+    if len(module_name) is not 0:
+        console.info('Data for the following modules will be plotted: {}.', set(processed_module_names))
+    else:
+        console.info('All modules will be plotted.')
+
+    host_list = df['host'].tolist()
+    for host in host_name:
+        if host not in host_list:
+            console.error('The host {} does not exist in your csv data. Exiting.', host)
+
+    if len(host_name) is not 0:
+        console.info('Data for the following hosts will be plotted: {}.', set(host_name))
+    else:
+        console.info('All hosts will be plotted.')
+
+    gpu_cpu_list = []
+    if gpu is True:
+        gpu_cpu_list.append(True)
+    if cpu is True:
+        gpu_cpu_list.append(False)
+
+    # here I split all data frames into the posible smallest segments
+    # this is necessary so we can plot all individually
+    split_df = df.groupby(['gpu', 'module', 'host'])
+
+    # here I initialize the list which will be plotted
+    df_list = []
+
+    for key, df in split_df:
+        if (
+            any(gpu in key for gpu in gpu_cpu_list) and
+            len(host_name) == 0 and
+            len(processed_module_names) == 0
+        ):
+            df_list.append(df)
+        elif (
+            any(gpu in key for gpu in gpu_cpu_list) and
+            any(host in key for host in host_name) and
+            len(processed_module_names) == 0
+        ):
+            df_list.append(df)
+        elif (
+            any(gpu in key for gpu in gpu_cpu_list) and
+            len(host_name) == 0 and
+            any(module in key for module in processed_module_names)
+        ):
+            df_list.append(df)
+
+        elif (
+            any(gpu in key for gpu in gpu_cpu_list) and
+            any(host in key for host in host_name) and
+            any(module in key for module in processed_module_names)
+        ):
+            df_list.append(df)
+
+    if len(df_list) == 0:
+        console.error(
+            'Your selections contained no Benchmarking Information.\n'
+            'Are you sure all your selections are correct?')
+    else:
+        console.info('A total of {} runs will be plotted.', len(df_list))
+
+    df = pd.concat(df_list)
+    fig = Figure()
+    FigureCanvas(fig)
+    ax = fig.add_subplot(111)
+    plot_over_group(df, plot_cores, ax=ax)
+
+    if output_name is None:
+        output_name = generate_output_name(output_type)
+    if output_type not in output_name:
+        output_name = '{}.{}'.format(output_name, output_type)
+    fig.savefig(output_name)
```

Comment:
Although `savefig` determines the filetype from the given file name, I would also explicitly pass it: `fig.savefig(output_name, type=output_type)`